src/controllers/authControllers.js

```diff
@@ -0,0 +1,157 @@
+const { User } = require('../models/user');
+const {
+  HttpError,
+  ctrlWrapper,
+  resizeImage,
+  sendEmail,
+} = require('../helpers');
+const generateAndSaveUserToken = require('../utils/generateAndSaveUserToken');
+const { verifyEmail } = require('../templates');
+
+const gravatar = require('gravatar');
+const path = require('path');
+const bcrypt = require('bcryptjs');
+const crypto = require('crypto');
+
+const avatarsDir = path.join(__dirname, '../../', 'public', 'avatars');
+
+const register = async (req, res) => {
+  const { email, password } = req.body;
+  const user = await User.findOne({ email });
+  if (user) {
+    throw new HttpError(409, 'Email already in use');
+  }
+
+  const hashPassword = await bcrypt.hash(password, 10);
+  const avatarURL = gravatar.url(email);
+  const verificationToken = crypto.randomUUID();
+
+  const registeredUser = await User.create({
+    ...req.body,
+    password: hashPassword,
+    avatarURL,
+    verificationToken,
+  });
+
+  await generateAndSaveUserToken(registeredUser);
+
+  await sendEmail(verifyEmail(email, verificationToken));
+
+  res.status(201).json({
+    user: {
+      email,
+      subscription: 'starter',
+    },
+  });
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+  const user = await User.findOne({ email });
+  if (!user) {
+    throw new HttpError(401, 'Email  or password invalid');
+  }
+
+  if (!user.verify) {
+    throw new HttpError(401, 'Email not verify');
+  }
+
+  const passwordCompare = await bcrypt.compare(password, user.password);
+  if (!passwordCompare) {
+    throw new HttpError(401, 'Email  or password invalid');
+  }
+
+  const newToken = await generateAndSaveUserToken(user);
+  res.json({
+    user: {
+      name: user.name,
+      email: user.email,
+    },
+    token: newToken,
+  });
+};
+
+const logout = async (req, res) => {
+  const { _id } = req.user;
+  await User.findByIdAndUpdate(_id, { token: '' });
+  res.status(204).json({ message: 'Logout susses' });
```

Comment:
```suggestion
  res.status(200).json({ message: 'Logout susses' });
```
якщо використовуєте статус 204, то в респонсі нічого не повернеться